src/packet_analysis/protocol/ip/IPBasedAnalyzer.cc

```diff
@@ -115,7 +115,9 @@ bool IPBasedAnalyzer::AnalyzePacket(size_t len, const uint8_t* data, Packet* pkt
 
 bool IPBasedAnalyzer::CheckHeaderTrunc(size_t min_hdr_len, size_t remaining, Packet* packet)
 	{
-	if ( packet->ip_hdr->PayloadLen() < min_hdr_len )
+	// If segment offloading or similar is enabled, the payload len will return 0.
+	// Thus, let's ignore that case.
+	if ( packet->ip_hdr->PayloadLen() && packet->ip_hdr->PayloadLen() < min_hdr_len )
```

Comment:
This is the worrisome part of the patch I guess. :-) I'm feeling quite unsure about this too.

 One thing I would suggest (and might have done that already) would be to look for more calls of `PayloadLen` and see if they deal with the zero case correctly. But that won't cover it. Another thought: can we add another check right after IP processing that bails out, without forwarding any data any further, if TSO is present?

Reply:
> Another thought: can we add another check right after IP processing that bails out, without forwarding any data any further, if TSO is present?

Well, that would need to be TCP/UDP/ICMP I guess, not IP.  And then it's probably not different from what we're already doing, is it?

Reply:
Well, the point of this patch is that we probably do want to process the data if TSO is enabled. So - we probably don't want to bail.

I will check all other occurrences of PayloadLen to check what they are doing - there are only a few.